src/backend/expungeservice/expungement_analyzer/analyzers/type_analyzer.py

```diff
@@ -0,0 +1,74 @@
+class TypeAnalyzer:
+
+    @staticmethod
+    def evaluate(charges):
+        for charge in charges:
+            TypeAnalyzer.__evaluate(charge)
+
+    @staticmethod
+    def __evaluate(charge):
+        if charge.acquitted():
+            charge.expungement_result.set_type_eligibility(True)
+            charge.expungement_result.set_reason('Eligible under 137.225(1)(b)')
+
+        elif TypeAnalyzer.__marijuana_ineligible(charge):
+            charge.expungement_result.set_type_eligibility(False)
+            charge.expungement_result.set_reason('Ineligible under 137.226')
+
+        elif TypeAnalyzer.__list_b_charge(charge):
+            charge.expungement_result.set_reason('Further Analysis Needed')
+
+        elif TypeAnalyzer.__ineligible_under_137_225_5(charge):
+            charge.expungement_result.set_type_eligibility(False)
+            charge.expungement_result.set_reason('Ineligible under 137.225(5)')
+
+        elif TypeAnalyzer.__possession_sched_1(charge):
+            charge.expungement_result.set_type_eligibility(True)
+            charge.expungement_result.set_reason('Eligible under 137.225(5)(C)')
+
+        elif charge.level == 'Felony Class C' or 'Misdemeanor' in charge.level:
+            charge.expungement_result.set_type_eligibility(True)
+            charge.expungement_result.set_reason('Eligible under 137.225(5)(b)')
+
+        elif charge.level == 'Felony Class B':
+            charge.expungement_result.set_reason('Further Analysis Needed')
+
+        else:
+            charge.expungement_result.set_reason('Examine')
+
+    @staticmethod
+    def __ineligible_under_137_225_5(charge):
+        return TypeAnalyzer.__crime_against_person(charge) or TypeAnalyzer.__traffic_crime(charge) or TypeAnalyzer.__felony_class_a(charge)
+
+    @staticmethod
+    def __crime_against_person(charge):
+        statute = int(charge.statute[0:6])
+        statute_ranges = (range(163305, 163480), range(163670, 163694), range(167008, 167108), range(167057, 167081))
+        return any(statute in statute_range for statute_range in statute_ranges)
```

Comment:
That may be true. I like the simplicity and readability that comes with using the `in` operator. If we do see performance issues later on we can run an analyzer to see where we need to optimize and go from there.

Reply:
Casting this to an int is going to cause issues. Portland's drug free zone variance statute has a letter in the third digit of the statute number. I'm thinking I will check with regex for any non numbers before casting; otherwise just return false. Will have to do that with the other slice [0:3] too.